### PR TITLE
BaSP-TG-76: Admins redux - Get

### PR DESCRIPTION
**Pull request by matescarabino**
- Create constants to GET admins
- Create actions to GET admins
- Create thunk to GET admins
- Create admin reducer
- Connect list component with store to GET admins
- Implement spinner while fetch is pending
- modify modalMessage to show correct error

QA:
There is no need to test until the other admin ticket is ready (TG-77)

### CODE REVIEW

**Review by Ansitox — CHANGES_REQUESTED** [review]
Everything ok, just a few commented lines that you should delete.

File: src/redux/admins/reducer.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    case GET_ADMINS_PENDING:
+      return {
+        ...state,
+        isLoading: true
+      };
+    case GET_ADMINS_SUCCESS:
+      return {
+        ...state,
+        isLoading: false,
+        error: '',
```
Comment (CristianBucci) [inline src/redux/admins/reducer.js:27]: No need to pass the error here!

File: src/redux/admins/reducer.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+const INITIAL_STATE = {
+  isLoading: false,
+  list: [],
+  error: '',
+  modalContent: { title: '', content: '' },
+  showModalMessage: false
+};
+
+const reducer = (state = INITIAL_STATE, action) => {
```
Comment (CristianBucci) [inline src/redux/admins/reducer.js:16]: This should be adminsReducer

File: src/redux/admins/constants.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const GET_ADMINS_PENDING = 'GET_ADMINS_PENDING;';
+export const GET_ADMINS_SUCCESS = 'GET_ADMINS_SUCCESS';
+export const GET_ADMINS_ERROR = 'GET_ADMINS_ERROR';
+
```
Comment (CristianBucci) [inline src/redux/admins/constants.js:4]: Delete this empty line!

File: src/redux/admins/reducer.js (hunk 1 above)
Comment (matescarabino) [reply src/redux/admins/reducer.js:27]: Done

File: src/redux/admins/reducer.js (hunk 2 above)
Comment (matescarabino) [reply src/redux/admins/reducer.js:16]: Done

File: src/redux/admins/constants.js (hunk 3 above)
Comment (matescarabino) [reply src/redux/admins/constants.js:4]: Done

**Review by CristianBucci — APPROVED** [review]

File: src/Components/Admins/index.js [hunk 4]
```diff
@@ -1,45 +1,46 @@
...
-  useEffect(async () => {
-    try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/admin`);
-      const data = await response.json();
-      setAdmins(data.data);
-    } catch (error) {
-      setModalContent({ title: 'ERROR!', content: `Could not GET admins! ${error.message}` });
-      setShowModalMessage(true);
-    }
+  useEffect(() => {
```
Comment (ignacioaristo) [inline src/Components/Admins/index.js:42]: You are doing this in line 32

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/Components/Admins/index.js (hunk 4 above)
Comment (matescarabino) [reply src/Components/Admins/index.js:42]: Done

**Review by nicolasblasco — APPROVED** [review]

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**